src/components/File/hooks/use-file-by-filepath.ts

```diff
@@ -107,6 +107,31 @@ const useFileByFilepath = () => {
     openFileByPath(absolutePath)
   }
 
+  const openAbsolutePath = async (filePath: string, optionalContentToWriteOnCreate?: string): Promise<void> => {
```

Comment:
> I don't think we need this function. It seems to be very similar to the openRelative except the check for whether its an absolute path.

The issue I run into is that openRelative takes the path and joins it with the vault directory. The issue is that the `filePath` state we store is the `absolutePath` which means if we want to easily open a file, we'd have to first remove the vaultDirectory section then call it. This is tedious for every call. Instead this function just simplifies the logic since our ipcHandler stores absolutePaths and we pass through our prop an absolutePath. I think this is a fine function.

> I think we should just have a generic function which opens path (both absolute and relative) and then we should perform the check for whether its an absolute path with an ipc handler and the node path.isAbsolute() function

Sure this could work. However, `path.isAbsolute` does not work properly on windows since files begin with a drive instead of `/`

Reply:
Ok fair enough. Have you tested this on windows?

Reply:
Ok fair enough. Have you tested on Windows to confirm that it doesn't work? Because from this it seems like node:path behaviour depends on the host machine: https://nodejs.org/api/path.html#windows-vs-posix